### PR TITLE
Update contribution guidelines

### PR DESCRIPTION
**Pull request by roypat**
## Changes

- Clarify guidance on using `.unwrap()`
- Correct statement about how many approvals are needed
- Remove former team members from CODEOWNERS/MAINTAINERS.md

## Reason

#808 

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following Developer
Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this
  PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet
  [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

______________________________________________________________________

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -119,6 +119,33 @@ Your contribution needs to meet the following standards:
   }
   ```
 
+- Similarly, avoid using `Option::unwrap`/`Result::unwrap`. Prefer propagating
```
Comment (xmarcalx) [inline CONTRIBUTING.md:122]: nit: I would remove "similarly", it does not scale If tomorrow we will need to add another condition which is similar but need another bullet point. We can simply consider this a list of rules. Start with "Avoid ..."

**Review by xmarcalx — CHANGES_REQUESTED** [review]
Thanks for the changes @roypat. Overall LGTM but i have a small suggestion which i passed on in the review.

File: CONTRIBUTING.md (hunk 1 above)
Comment (roypat) [reply CONTRIBUTING.md]: Thanks! Updated :)

**Review by xmarcalx — APPROVED** [review]
LGTM, thanks for the changes.

**Pull request merged**